src/hooks/useFetch/index.ts

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from "react";
+import { isProduction } from "utils/environmentVariables";
+import { leaveBreadcrumb } from "utils/errorReporting";
+/**
+ * `useFetch` is a custom hook that downloads json from a given URL.
+ * It returns an object with the following properties:
+ * - isLoading: a boolean that is true while the log is being downloaded
+ * - data: the log file as an array of strings
+ * - error: an error message if the download fails
+ *
+ */
+const useFetch = <T extends object>(
+  url: string,
+  options?: {
+    skip?: boolean;
+  }
+) => {
+  const [data, setData] = useState<T | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    leaveBreadcrumb("useFetch", { url }, "request");
+    const req = new Request(url, { method: "GET" });
+    const abortController = new AbortController();
+
+    if (!options?.skip) {
+      setIsLoading(true);
```

Comment:
hmm instead of doing `setIsLoading` here, it might be possible to set it on initialization of state on `L20` based on the value of `skip`

Reply:
This actually isn't true because if we start by skipping the query and then setting `skip=false` isLoading won't be set back to true. I added a test to assert this.